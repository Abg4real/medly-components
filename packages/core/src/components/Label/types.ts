import { WithThemeProp } from '@medly-components/utils';

export interface Props extends WithThemeProp {
    labelPosition?: 'top' | 'start' | 'end' | 'bottom';
    required?: boolean;
}
