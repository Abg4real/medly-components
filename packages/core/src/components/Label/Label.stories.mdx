import { Label } from './Label';
import { defaultTheme } from '@medly-components/theme';
import { Preview, Story, Meta, Props } from '@storybook/addon-docs/blocks';
import * as stories from './Label.stories';

<Meta title="Core/Label" component={Label} />

# Label

### Basic Label

<Preview withToolbar>
    <Story name="Basic">{stories.Basic}</Story>
</Preview>

### Props

<Props of={Label} />

### LabelTheme

<Props of={stories.ThemeInterface} />

### LabelTheme Defaults

<code>{JSON.stringify(defaultTheme.label, null, 2)}</code>
