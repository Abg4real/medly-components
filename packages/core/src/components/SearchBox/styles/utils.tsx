import { SvgIcon } from '@medly-components/icons';
import { css } from '@medly-components/utils';
import { Props } from '../types';

export const getSearchBoxSize = ({ theme, size }: Props) => theme.searchBox.sizes[size].height;

export const getBorderAndBoxShadow = (color: string, boxShadow: string) => {
    return css`
        border-color: ${color};
        box-shadow: ${boxShadow};
    `;
};

export const getIconWrapperStyle = ({ theme, size }: Props) => {
    return css`
        width: ${getSearchBoxSize};
        min-width: ${getSearchBoxSize};
        height: auto;
        display: inline-flex;
        align-items: center;
        justify-content: center;
        ${SvgIcon} {
            cursor: pointer;
            font-size: ${theme.searchBox.sizes[size].iconSize};
        }
    `;
};
