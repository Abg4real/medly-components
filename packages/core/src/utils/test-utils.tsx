import { defaultTheme } from '@medly-components/theme';
import React from 'react';
import { render, RenderResult } from 'react-testing-library';
import { ThemeProvider } from './styled';

const AllTheProviders: React.FunctionComponent = props => (
    <ThemeProvider theme={defaultTheme}>
        <>{props.children}</>
    </ThemeProvider>
);

const customRender = (ui: React.ReactElement<any>, options?: object): RenderResult => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from 'react-testing-library';

// override render method
export { customRender as render };
