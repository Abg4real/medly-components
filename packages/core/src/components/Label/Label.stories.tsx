import { defaultTheme, FontSizes, FontWeights, LabelTheme } from '@medly-components/theme';
import { boolean, color, select } from '@storybook/addon-knobs';
import React from 'react';
import { Label } from './Label';

const sizeOptions: FontSizes[] = ['S1', 'M1', 'M2', 'M3', 'L1', 'L2', 'L3', 'L4'];
const weightOptions: FontWeights[] = ['Light', 'Normal', 'Strong'];
const defaultColor = defaultTheme.font.defaultColor;

export const ThemeInterface = (props: LabelTheme): any => null;

export const Basic = () => (
    <Label
        showPointer={boolean('Show Pointer', false)}
        required={boolean('Show Required asterisk', false)}
        labelColor={color('Color', defaultColor)}
        labelSize={select('Label Size', sizeOptions, 'L1')}
        labelWeight={select('Label Weight', weightOptions, 'Normal')}
    >
        Top 5
    </Label>
);
