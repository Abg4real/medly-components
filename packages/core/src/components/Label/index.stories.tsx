import { defaultTheme, FontSizes, FontWeights } from '@medly-components/theme';
import { boolean, color, select } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';
import Label from './Label';

const sizeOptions: FontSizes[] = ['S1', 'M1', 'M2', 'M3', 'L1', 'L2', 'L3', 'L4'];
const weightOptions: FontWeights[] = ['Light', 'Normal', 'Strong'];
const defaultColor = defaultTheme.font.defaultColor;

storiesOf('Core', module).add('Label', () => (
    <Label
        showPointer={boolean('Show Pointer', false)}
        required={boolean('Show Required asterisk', false)}
        textColor={color('Color', defaultColor)}
        textSize={select('Text Size', sizeOptions, 'L1')}
        textWeight={select('Text Weight', weightOptions, 'Normal')}
    >
        Top 5
    </Label>
));
