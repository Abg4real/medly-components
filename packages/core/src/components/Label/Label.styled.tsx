import { defaultTheme } from '@medly-components/theme';
import { css, oppositePositionalspacing, styled } from '@medly-components/utils';
import { Props } from './types';

const asterisk = () => css`
    ::after {
        color: red;
        content: ' *';
    }
`;

const pointerCursor = () => css`
    cursor: pointer;
`;

export const LabelStyled = styled('label')<Props>`
    margin: ${({ theme, labelPosition }) => oppositePositionalspacing(labelPosition, theme.spacing.S)};
    color: ${({ theme }) => theme.label.color};
    ${({ required }) => required && asterisk()}
    ${({ showPointer }) => showPointer && pointerCursor()}
`;

LabelStyled.defaultProps = {
    theme: defaultTheme
};
