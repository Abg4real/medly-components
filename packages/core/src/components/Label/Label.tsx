import { WithStyle } from '@medly-components/utils';
import React, { FC } from 'react';
import { LabelStyled } from './Label.styled';
import { Props } from './types';

export const Label: FC<Props> & WithStyle = React.memo(
    React.forwardRef((props, ref) => (
        <LabelStyled ref={ref} {...props}>
            {props.children}
        </LabelStyled>
    ))
);

Label.defaultProps = {
    labelPosition: 'left',
    required: false
};
Label.displayName = 'Label';
Label.Style = LabelStyled;
