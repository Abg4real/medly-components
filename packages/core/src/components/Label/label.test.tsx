import { TestUtils } from '@medly-components/utils';
import React from 'react';
import Label from './Label';

describe('Label component', () => {
    it('should render correctly with default props', () => {
        const { container } = TestUtils.render(<Label>Label</Label>);
        expect(container).toMatchSnapshot();
    });

    it('should render correctly with all the props given', () => {
        const { container } = TestUtils.render(
            <Label required showPointer labelPosition="top" labelSize="M2" labelWeight="Light" labelColor="pink">
                Demo Label
            </Label>
        );
        expect(container).toMatchSnapshot();
    });
});
