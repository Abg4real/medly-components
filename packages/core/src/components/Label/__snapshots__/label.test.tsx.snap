// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label component should render correctly with all the props given 1`] = `
.c0 {
  color: pink;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  margin: 0px 0px 8px 0px;
  cursor: pointer;
}

.c0::after {
  color: red;
  content: ' *';
}

<div>
  <label
    class="c0"
    required=""
  >
    Demo Label
  </label>
</div>
`;

exports[`Label component should render correctly with default props 1`] = `
.c0 {
  color: #435465;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  margin: 0px 8px 0px 0px;
  cursor: default;
}

<div>
  <label
    class="c0"
  >
    Label
  </label>
</div>
`;
